eZ/Publish/Core/Limitation/Tests/LanguageLimitationTypeTest.php

```diff
@@ -0,0 +1,467 @@
+                'object' => $this->getVersionInfo('fra-FR'),
+                'expected' => false,
+            ),
+            // VersionInfo, no access, NONE of limitationValues is in VersionInfo->languageCodes
+            array(
+                'limitation' => new LanguageLimitation(array('limitationValues' => array('eng-EN', 'pol-PL'))),
+                'object' => $this->getVersionInfo('fra-FR', ['nor-NO']),
+                'expected' => false,
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerForTestEvaluate
+     */
+    public function testEvaluate(
+        LanguageLimitation $limitation,
+        ValueObject $object,
+        $expected
+    ) {
+        // Need to create inline instead of depending on testConstruct() to get correct mock instance
+        $limitationType = $this->testConstruct();
+
+        $userMock = $this->getUserMock();
+        $userMock
+            ->expects($this->never())
+            ->method($this->anything());
+
+        $value = $limitationType->evaluate(
+            $limitation,
+            $userMock,
+            $object
+        );
+
+        self::assertInternalType('boolean', $value);
+        self::assertEquals($expected, $value);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerForTestEvaluateInvalidArgument()
+    {
+        return array(
+            // invalid limitation
+            array(
+                'limitation' => new ObjectStateLimitation(),
+                'object' => new ContentInfo(),
+                'targets' => null,
+                'persistence' => array(),
+            ),
+            // invalid object
+            array(
+                'limitation' => new LanguageLimitation(),
+                'object' => new ObjectStateLimitation(),
+                'targets' => array(new Location()),
+                'persistence' => array(),
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerForTestEvaluateInvalidArgument
+     * @expectedException \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    public function testEvaluateInvalidArgument(
+        Limitation $limitation,
+        ValueObject $object,
+        $targets,
+        array $persistenceLocations
+    ) {
+        // Need to create inline instead of depending on testConstruct() to get correct mock instance
+        $limitationType = $this->testConstruct();
+
+        $userMock = $this->getUserMock();
+        $userMock
+            ->expects($this->never())
+            ->method($this->anything());
+
+        $persistenceMock = $this->getPersistenceMock();
+        $persistenceMock
+            ->expects($this->never())
+            ->method($this->anything());
+
+        $v = $limitationType->evaluate(
+            $limitation,
+            $userMock,
+            $object,
+            $targets
+        );
+        var_dump($v); // intentional, debug in case no exception above
```

Comment:
ehm, please no :) If you must, then `self::fail` with a result of `var_export`, but I wouldn't go this way anyway.